wallet-new/src/Cardano/Wallet/API/V1/LegacyHandlers/Wallets.hs

```diff
@@ -186,3 +198,95 @@ updateWallet wid WalletUpdate{..} = do
+newExternalWallet NewExternalWallet{..} = do
+    let newWalletHandler CreateWallet  = createNewExternalWallet
+        newWalletHandler RestoreWallet = error "Restore external wallet, TODO"
+    walletMeta <- V0.CWalletMeta <$> pure newewalName
+                                 <*> migrate newewalAssuranceLevel
+                                 <*> pure 0
+    single <$> do
+        v0wallet <- newWalletHandler newewalOperation walletMeta newewalExtPubKey
+        ss <- V0.askWalletSnapshot
+        addWalletInfo ss v0wallet
+
+-- | Creates new external wallet.
+--
+-- There's no spending password, because it's assumed that
+-- extenal wallet has its own security mechanism.
+-- For example, in Ledger Nano S device it's 4 digits PIN-code.
+--
+-- There's no backup phrase as well, because it's assumed that
+-- external wallet already received backup phrase, created
+-- a secret key based on it and stored this key.
+createNewExternalWallet
+    :: ( MonadThrow m
+       , V0.MonadBlockchainInfo m
+       , V0.MonadWalletLogic ctx m
+       )
+    => V0.CWalletMeta
+    -> Text
+    -> m V0.CWallet
+createNewExternalWallet walletMeta encodedExtPubKey = do
+    -- 'encodedExtPubKey' is Base58-encoded binary string that contains extended public key
+    -- of external wallet.
+    publicKey <- makePublicKeyFrom encodedExtPubKey
+    -- Add this public key in the 'public.key' file. Public key will be used during
+    -- synchronization with the blockchain.
+    addPublicKey publicKey
+    walletId <- makeWalletIdFrom publicKey
+    let isReady = True -- A brand new wallet doesn't need syncing with the blockchain.
+
+    -- Create new external wallet.
+    -- This is safe: if the client will try to create an external wallet from the same
+    -- extended public key - error will be thrown.
+    V0.CWallet{..} <- V0.createWalletSafe walletId walletMeta isReady
+
+    -- Add initial account in this external wallet.
+    let accountMeta    = V0.CAccountMeta { caName = "Initial account" }
+        accountInit    = V0.CAccountInit { caInitWId = cwId, caInitMeta = accountMeta }
+        includeUnready = True
+    void $ V0.newExternalAccountIncludeUnready includeUnready accountInit
+
+    db <- V0.askWalletDB
+    removeHistoryCache db walletId
+    -- BListener checks current syncTip before applying update,
+    -- thus setting it up to date manually here
+    withStateLockNoMetrics HighPriority $ \tip -> setWalletSyncTip db walletId tip
+    V0.getWallet walletId
+  where
+    makePublicKeyFrom encodedXPub = do
+        let rawExtPubKey = decodeBase58 bitcoinAlphabet . encodeUtf8 $ encodedXPub
+        when (isNothing rawExtPubKey) $
+            throwM (UnknownError "Ext public key is not in proper Base58-form.")
+
+        let extPubKey = xpub $ fromJust rawExtPubKey
+        when (isLeft extPubKey) $ do
+            let (problem:_) = lefts [extPubKey]
+            throwM (UnknownError $ "Invalid ext public key: " <> toText problem)
```

Comment:
As above, please add the error case to the error sum type instead of using the `UnknownError` constructor.